feat: adiciona handling de logs

carlohcs opened the pull request:
# Descrição

- Adiciona logs ao acessar algum recurso;

## Evidências

**Salvando em arquivo de texto**

<img width="2056" alt="Captura de Tela 2025-01-18 às 16 57 22" src="https://github.com/user-attachments/assets/aa2e168c-8c9d-4d07-8af0-491475425d37" />

**Salvando no banco de dados**
![image](https://github.com/user-attachments/assets/6a11ea2f-23e2-4c43-9118-fbaa24363e56)

**Salvando no banco de dados - Com e sem usuário"
![image](https://github.com/user-attachments/assets/c654fbc6-2211-4c9f-acd7-f3add886ce2e)

carlohcs committed:
feat: adiciona handling de logs

github-actions[bot] committed:
[GitHub Actions] Autogenerated JaCoCo coverage badge

lcalves commented on src/main/java/br/com/alfac/videostudio/infra/config/WebConfig.java:
Apagar classe

carlohcs replied:
Done!

lcalves commented on src/main/java/br/com/alfac/videostudio/infra/persistence/LogEntity.java:
```diff
@@ -0,0 +1,73 @@
+public class LogEntity implements Serializable {
+
+    @Id
+    private Long id;
```
adicionar o   @GeneratedValue(strategy = GenerationType.IDENTITY)

carlohcs replied:
Done!

lcalves commented on src/main/java/br/com/alfac/videostudio/infra/persistence/LogRepository.java:
```diff
@@ -0,0 +1,6 @@
+import org.springframework.data.jpa.repository.JpaRepository;
+import java.util.UUID;
+
+public interface LogRepository extends JpaRepository<LogEntity, UUID> {
```
Vai dar problema com o tipo da key da entity. trocar de UUID para Long

carlohcs replied:
Done!

lcalves commented on src/main/resources/db/V2_1_create_log_table.sql:
```diff
@@ -0,0 +1,6 @@
+CREATE TABLE log (
+    id BIGINT PRIMARY KEY,
```
adicionar o auto_increment para gerar id automatico

carlohcs replied:
Done!

lcalves commented on src/main/java/br/com/alfac/videostudio/infra/config/WebConfig:
apagar arquivo

carlohcs replied:
Done!

carlohcs committed:
feat: habilita o funcionamento do banco de dados para log

- aplica correções do pull request;
- remove arquivos desnecessários;
- corrige configs do banco de dados;
- simplifica algumas configs de envs;

github-actions[bot] committed:
[GitHub Actions] Autogenerated JaCoCo coverage badge

carlohcs committed:
feat: habilita o funcionamento do banco de dados para log

- aplica correções do pull request;
- remove arquivos desnecessários;
- corrige configs do banco de dados;
- simplifica algumas configs de envs;

github-actions[bot] committed:
[GitHub Actions] Autogenerated JaCoCo coverage badge
